### PR TITLE
[Merged by Bors] - 0.10 shader features

### PR DESCRIPTION
**Pull request by cart**

### CODE REVIEW

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 1]
```diff
@@ -686,6 +686,87 @@ app from the app to send it between the main thread and the rendering thread. Th
...
+
+```rust
+for (var i: u32 = 0u; i < #{MAX_DIRECTIONAL_LIGHTS}; i = i + 1u) {
+}
+```
+
+They can also be defined inline in shaders:
+
+```rust
+#define MAX_DIRECTIONAL_LIGHTS 10
```
Comment (LiamGallagher737) [inline content/news/2023-03-04-bevy-0.10/index.md:721]: Here the 10 is not clear to me, is this a default value?

Comment (cart) [reply content/news/2023-03-04-bevy-0.10/index.md:721]: Its just an arbitrary value chosen to illustrate what setting values look like. Its equivalent to `let max_directional_lights = 10` in rust.

Comment (LiamGallagher737) [reply content/news/2023-03-04-bevy-0.10/index.md:721]: I get that. What I don't understand is how this fits in with passing constant values in from rust. If I passed in a value of 15 using `ShaderDefVal::Int("MAX_DIRECTIONAL_LIGHTS".to_string(), 15)` would that override the value defined in the shader?

Comment (cart) [reply content/news/2023-03-04-bevy-0.10/index.md:721]: Ahh yeah the precedence order is worth clarifying. Good call!

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 2]
```diff
@@ -686,6 +686,87 @@ app from the app to send it between the main thread and the rendering thread. Th
 only valid to do after all the plugin build methods have been called, because any plugin may
 want to modify the rendering sub app.
 
+## ShaderDef Values
+
+<div class="release-feature-authors">authors: @mockersf</div>
+
+Bevy's shader processor now supports ShaderDefs with values. This allows developers to pass constant values into their shaders:
```
Comment (alice-i-cecile) [inline content/news/2023-03-04-bevy-0.10/index.md:693]: ShaderDefs should probably have a link.

**Review by LiamGallagher737 — APPROVED** [review]

**Pull request closed without merging**